nio/schemas.py

```diff
@@ -1269,18 +1269,22 @@ class Schemas(object):
             "content": {
                 "type": "object",
                 "properties": {
-                    "tags": {"type:": "object"},
-                    "properties": {
-                        "order": {"type": "float"}
+                    "tags": {
+                        "type:": "object",
+                        "patternProperties": {
+                            r".*": {
+                                "type": "object",
+                                "properties": {
+                                    "order": {"type": "number"}
```

Comment:
The order is a required property, the pattern property here needs a required array with "order" in it.

The rest looks good. After you add this we can merge.

Reply:
I am not sure if it is really required. I get "lowpriority" tags with no order. Just an example out of a recorded session:

```json
 { 'events': [ { 'content': { 'tags': { 'm.lowpriority': { }}},          
 'type': 'm.tag'},
{ 'content': { 'event_id': '$xxxxxx:my.host'},
'type': 'm.fully_read'}]}
```

Reply:
... but there is no description about these tags. My sample is sent by an synapse server 0.99.2.

Reply:
just for proving: this is a "favourite" tag from the same session:

```
{ 'events': [ { 'content': { 'tags': { 'm.favourite': { 'order': 0.125}}},
        'type': 'm.tag'},
{ 'content': { 'event_id': '$yyyyy:my.host'},
        'type': 'm.fully_read'}]},
```

Reply:
Ah: spec 13.18.1:

> . If a room has a tag without an order key then it should appear after the rooms with that tag that have an order key.

Reply:
Oh, i must have miss-remembered something, sorry about that. All good then.